FirebaseStorageSwift/Sources/SwiftApiExtension.swift

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseStorage
+
+/// getResultCallback generates a closure that returns a Result type from a closure that returns an
+/// optional type and Error.
+private func getResultCallback<T>(
+  completion: @escaping (Result<T, Error>) -> Void
+) -> (_: T?, _: Error?) -> Void {
+  return { (metadata: T?, error: Error?) -> Void in
+    guard let metadata = metadata else {
```

Comment:
Optional: This might  be easier to follow if you split this into two guard statement that are not nested.

Reply:
Good idea! This is an example that is better without using `guard` statements - with a plain old if then elseif, else, ....

Done.